pfcwd/main.py

```diff
@@ -32,272 +53,438 @@
+                config_list.append(line)
+            table.append([port] + config_list)
+
+        poll_interval = self.config_db.get_entry(
+            CONFIG_DB_PFC_WD_TABLE_NAME, 'GLOBAL'
+        ).get('POLL_INTERVAL')
+
+        current_ns = self.multi_asic.current_namespace
+        asic_ns = \
+            "" if current_ns is None or current_ns == "" else " on {}".format(
+                current_ns
+            )
+        if poll_interval is not None:
+            click.echo(
+                "Changed polling interval to {} ms{}".format(
+                    poll_interval, asic_ns
+                )
+            )
+
+        big_red_switch = self.config_db.get_entry(
+            CONFIG_DB_PFC_WD_TABLE_NAME, 'GLOBAL'
+        ).get('BIG_RED_SWITCH')
+
+        if big_red_switch is not None:
+            click.echo("BIG_RED_SWITCH status is {}{}".format(
+                big_red_switch, asic_ns
+            ))
+
+        self.table += table
+
+    def config(self, ports):
+        del self.table[:]
+        self.collect_config(ports)
+        click.echo(tabulate(
+            self.table, CONFIG_HEADER, stralign='right', numalign='right',
+            tablefmt='simple'
+        ))
+
+    @multi_asic_util.run_on_multi_asic
+    def start(self, action, restoration_time, ports, detection_time):
+        if os.geteuid() != 0:
+            exit("Root privileges are required for this operation")
+        allowed_strs = ['ports', 'all', 'detection-time']
+        configdb = swsssdk.ConfigDBConnector()
+        configdb.connect()
+        countersdb = swsssdk.SonicV2Connector(host='127.0.0.1')
+        countersdb.connect(countersdb.COUNTERS_DB)
+
+        all_ports = get_all_ports(
+            self.db, self.multi_asic.current_namespace,
+            self.multi_asic.display_option
+        )
+        allowed_strs = allowed_strs + all_ports
+        for p in ports:
+            if p not in allowed_strs:
+                raise click.BadOptionUsage(
+                    "Bad command line format. Try 'pfcwd start --help' for"
+                    "usage"
+                )
+
+        if len(ports) == 0:
+            ports = all_ports
+
+        pfcwd_info = {
+            'detection_time': detection_time,
+        }
+        if action is not None:
+            pfcwd_info['action'] = action
+        if restoration_time is not None:
+            pfcwd_info['restoration_time'] = restoration_time
+        else:
+            pfcwd_info['restoration_time'] = 2 * detection_time
+            click.echo(
+                "restoration time not defined; default to 2 times"
+                "detection time: {} ms".format(2 * detection_time)
+            )
+
+        for port in ports:
+            if port == "all":
+                for p in all_ports:
+                    self.config_db.mod_entry(
+                        CONFIG_DB_PFC_WD_TABLE_NAME, p, None
+                    )
+                    self.config_db.mod_entry(
```

Comment:
since we are not specifying the namespace option for config commands, aren't we updating the global db? How does 'show config' for a namespace work with this approach?

Reply:
for all the 'config' commands, the command is for all namespaces, and the DB within the namespace is updated. 

By default, 'show config'  will fetch from namespaces that have external ports.

This logic emulates a 'single' switch interface.